src/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Test, TestingModule } from '@nestjs/testing';
+import { AuthService } from './auth.service';
+import { PassportModule } from '@nestjs/passport';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { AuthToken } from './auth-token.entity';
+import { User } from '../users/user.entity';
+import { UsersModule } from '../users/users.module';
+import { Identity } from '../users/identity.entity';
+import { LoggerModule } from '../logger/logger.module';
+
+describe('AuthService', () => {
+  let service: AuthService;
+  let user: User;
+  let authToken: AuthToken;
+
+  beforeEach(async () => {
+    user = {
+      authTokens: [],
+      createdAt: new Date(),
+      displayName: 'hardcoded',
+      id: '1',
+      identities: [],
+      ownedNotes: [],
+      updatedAt: new Date(),
+      userName: 'Testy',
+    };
+
+    authToken = {
+      accessTokenHash: '',
+      createdAt: new Date(),
+      id: 1,
+      identifier: 'testIdentifier',
+      keyId: 'abc',
+      lastUsed: null,
+      user: null,
+      validUntil: null,
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        AuthService,
+        {
+          provide: getRepositoryToken(AuthToken),
+          useValue: {},
+        },
+      ],
+      imports: [PassportModule, UsersModule, LoggerModule],
+    })
+      .overrideProvider(getRepositoryToken(AuthToken))
+      .useValue({
+        findOne: (): AuthToken => {
+          return {
+            ...authToken,
+            user: user,
+          };
+        },
+        save: async (entity: AuthToken) => {
+          if (entity.lastUsed === undefined) {
+            expect(entity.lastUsed).toBeUndefined();
+          } else {
+            expect(entity.lastUsed).toBeLessThanOrEqual(new Date().getTime());
+          }
+          return entity;
+        },
+        remove: async (entity: AuthToken) => {
+          expect(entity).toEqual({
+            ...authToken,
+            user: user,
+          });
+        },
+      })
+      .overrideProvider(getRepositoryToken(Identity))
+      .useValue({})
+      .overrideProvider(getRepositoryToken(User))
+      .useValue({
+        findOne: (): User => {
+          return {
+            ...user,
+            authTokens: [authToken],
+          };
+        },
+      })
+      .compile();
+
+    service = module.get<AuthService>(AuthService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('checkPassword', async () => {
```

Comment:
The names of unit tests should generally form a (somewhat) complete sentence. In [the Nest example](https://docs.nestjs.com/fundamentals/testing#unit-testing) the `describe` and `it` calls result in `CatsController find all should return an array of cats`.
Could you change these names into something similar?

Reply:
fixed